php/libraries/NDB_BVL_InstrumentStatus_ControlPanel.class.inc

```diff
@@ -104,11 +104,12 @@ class NDB_BVL_InstrumentStatus_ControlPanel extends NDB_BVL_InstrumentStatus
         }
 
         if (empty($this->subtest)) {
-            $hasAccess = $this->_hasAccess();
-
-            if ($this->getDataEntryStatus() == null && $hasAccess) {
-                $this->setDataEntryStatus('In Progress');
-            }
+            $status = new \NDB_BVL_InstrumentStatus;
+            $status->select($this->tpl_data['commentID']);
+            $this->setDataEntryStatus(
+                $status->_status['Data_entry']
+                ?? 'In Progress'
+            );
 
```

Comment:
why was the access check removed ?

Reply:
@ridz1208 the hasAccess check didn't make sense when looking at the previous code and I thought to remove it because it seems unnecessary. Should I add it back?

Reply:
I think the reason it's there is if the instrument has specific permissions and someone without these permissions clicks on it, it would not change the instrument to in progress

Reply:
@ridz1208 well now the code shouldn't change the instrument to "in progress" when viewing the instrument. So maybe it's unnecessary now?

Reply:
which is always the case on first access.

so with your code, if I dont have access to instrument X, but i click on it in the instrument list, it will still set it to in progress if it's the first time the instrument is accessed. so it's still an issue

Reply:
@ridz1208 I added it back.